src/main/java/com/massivedisaster/bintraydeployautomator/Automator.java

```diff
@@ -20,56 +22,85 @@
      *
      * @param args command line arguments.
      */
-    public static void main(String args[]) {
+    public static void main(String args[]) throws IOException {
 
         ProjectConnection gradleConnection = null;
+        ProgressBar pb = null;
+        String outputFile = null;
         try {
-            Pair<String, String> auth = CommandLineUtils.commandLineArgs(args);
+            Arguments auth = CommandLineUtils.commandLineArgs(args);
 
             // Get configuration from .json.
             Configuration configuration = Configuration.parseConfiguration("configuration.json");
 
-            configuration.setBintrayUsername(auth.getKey());
-            configuration.setBintrayKey(auth.getValue());
+            List<String> modules = configuration.getModules();
+
+            configuration.setBintrayUsername(auth.getUser());
+            configuration.setBintrayKey(auth.getKey());
+            configuration.setVerbose(auth.isVerbose());
+            outputFile = auth.getLogFile();
+            if (outputFile != null) {
+                File output = new File(outputFile);
+                if (output.exists()) {
+                    output.delete();
+                }
+                pb = new ProgressBar("BintrayDeployAutomator-" + com.massivedisaster.bintraydeployautomator.utils.Version.VERSION, modules.size() * 2);
+            }
 
             gradleConnection = GradleConnector.newConnector()
                     .forProjectDirectory(new File(configuration.getBasePath()))
                     .connect();
 
-            // Clean build and deploy all projects.
-            rebuildAndBintrayDeploy(gradleConnection, configuration);
+            ConsoleUtils.DrawInConsoleBox("Start process");
+
+            if (pb != null) pb.start();
+            //clean and build
+            for (String module : modules) {
+                if (pb != null) pb.setExtraMessage("Clean and building module " + module + "...");
+                GradleUtils.runGradle(gradleConnection, outputFile, new String[]{module + ":clean", module + ":build"}, configuration.getArguments());
+                if (pb != null) pb.step();
+            }
+
+            //clean and build
+            for (String module : modules) {
+                if (pb != null) pb.setExtraMessage("Uploading to bintray module " + module + "...");
```

Comment:
the check for pb != null was made above. we should't need this here.

Reply:
pb is only created when user adds the file to write the console output.